sentry-android-core/src/main/java/io/sentry/android/core/SendCachedEnvelopeIntegration.java

```diff
@@ -81,6 +83,15 @@ private synchronized void sendCachedEnvelopes(
       return;
     }
 
+    // in case there's rate limiting active, skip processing
+    final @Nullable RateLimiter rateLimiter = hub.getRateLimiter();
+    if (rateLimiter != null && rateLimiter.isActiveForCategory(DataCategory.All)) {
```

Comment:
`m`: I think we probably also have to add this to `EnvelopeFileObserver`, so if the hybrid SDKs are writing to the outbox folder we're also not trying to send envelopes if we're rate-limited right now